plugins/action/k8s_info.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2012-2014, Michael DeHaan <michael.dehaan@gmail.com>
+# Copyright (c) 2017, Toshio Kuratomi <tkuraotmi@ansible.com>
+# Copyright (c) 2020, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import traceback
+
+from ansible.module_utils._text import to_text
+from ansible.plugins.action import ActionBase
+from ansible.errors import AnsibleError
+
+
+class ActionModule(ActionBase):
+
+    TRANSFERS_FILES = True
+
+    def _ensure_invocation(self, result):
+        # NOTE: adding invocation arguments here needs to be kept in sync with
+        # any no_log specified in the argument_spec in the module.
+        # This is not automatic.
+        # NOTE: do not add to this. This should be made a generic function for action plugins.
+        # This should also use the same argspec as the module instead of keeping it in sync.
+        if 'invocation' not in result:
+            if self._play_context.no_log:
+                result['invocation'] = "CENSORED: no_log is set"
+            else:
+                # NOTE: Should be removed in the future. For now keep this broken
+                # behaviour, have a look in the PR 51582
+                result['invocation'] = self._task.args.copy()
+                result['invocation']['module_args'] = self._task.args.copy()
+
+        return result
+
+    def run(self, tmp=None, task_vars=None):
+        ''' handler for k8s options '''
+        if task_vars is None:
+            task_vars = dict()
+
+        result = super(ActionModule, self).run(tmp, task_vars)
+        del tmp  # tmp no longer has any effect
+
+        new_module_args = self._task.args.copy()
```

Comment:
For `k8s_info` specifically there aren't nested parameters accepted, but for anything that supports the `resource_definition` parameter it could be a concern